Assets/Gameplay/Core/Deployer.cs

```diff
@@ -23,16 +24,30 @@ public Deployer(GameObjectFactory gameObjectFactory, Arena arena, GameplayHUD ga
         public Entity DeployCard(CardType cardType, Team team, int laneIdx)
         {
             var card = GameObjectFactory.CreateCard(cardType, team);
-
+            
             var lane = Arena.Lanes[laneIdx];
             lane.AddEntity(card, team);
 
             var damageable = card.GetComponent<IDamageable>();
+            
             if (damageable != null)
             {
                 GameplayHUD.CreateHealthBar(damageable, team, card.transform);
             }
 
+            var behaviour = card.GetComponent<MultiDeployBehaviour>();
+
+            for (var i = 0; i < behaviour.count - 1; i++)
```

Comment:
A lógica está certinha, meninas! Parabéns!

A única crítica construtiva que tenho é de que a lógica está um pouco duplicada e poderia ser modularizada. Podíamos ter uma segunda função com a lógica de criação da carta e chamá-la pra carta original e dentro do for de multideploy.

Só pra ilustrar o problema, percebem que tem um `GameObjectFactory.CreateCard`, `lane.AddEntity` e `GameplayHUD.CreateHealthBar` duplicados? Isso é porque a lógica de criação da carta está duplicada. Daí a gente corre o risco de, ao modificar essa lógica ali pra carta original, esquecer de modificar/duplicar na parte do multideploy. Essa é a principal desvantagem, por isso acho que seria legal modularizar essa lógica e só chamar a função nos dois lugares (no começo pra carta original e no for do multideploy).

Acham que faz sentido?

Reply:
Faz sentindo sim.

Fizemos uma alteração no commit c82d6b87718e29365f7d9f0d7ec00c4d917dbd72 que seguiu mais ou menos essa ideia que você deu.